src/js/angular/sparql-editor/controllers.js

```diff
@@ -300,6 +292,78 @@ function SparqlEditorCtrl($scope, $location, $jwtAuth, $repositories, toastr, $t
     // ================================
     // =     Setup output handlers    =
     // ================================
+
+    /**
+     * Handles the "query" event emitted by the ontotext-yasgui. The event is fired immediately before sending the
+     * request and the request object can be altered here, and it will be sent with these changes.
+     * @param {object} queryResponseEvent - the event payload containing the query and the request object.
+     */
+    function queryHandler(queryResponseEvent) {
+        updateRequestHeaders(queryResponseEvent.request);
+        changeEndpointByQueryType(queryResponseEvent.queryMode, queryResponseEvent.request);
+        const pageNumber = queryResponseEvent.request._data['pageNumber'] ? parseInt(queryResponseEvent.request._data['pageNumber']) : undefined;
```

Comment:
I wonder, can we have a request object to some model mapper? As far as I can see, do a lot of checks here and there for the properties pageSize, offset, etc. So it might be convenient to have these in a mapper returning some internal representation of this request object. You can check here `src/js/angular/rest/mappers/saved-query-mapper.js` I have created some simple mapping functions.